.github/workflows/build-syncthing.yaml

```diff
@@ -0,0 +1,73 @@
+name: Build Syncthing
+
+on:
+  pull_request:
+  push:
+
+env:
+  GO_VERSION: "1.19.2"
+  GO386: softfloat
+  GOARM: "5"
+  GOMIPS: softfloat
+
+jobs:
+
+  build-windows:
+    runs-on: windows-latest
+    name: Build and test on Windows
+    steps:
+      - name: Set git to use LF
+        # Without this, the checkout will happen with CRLF line endings,
+        # which is fine for the source code but messes up tests that depend
+        # on data on disk being as expected. Ideally, those tests should be
+        # fixed, but not today.
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
+
+      - uses: actions/checkout@v3
+        # `fetch-depth: 0` because we want to check out the entire repo
+        # including tags and branches, not just the latest commit which
+        # lacks version info.
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v3
+        # `cache: true` gives us automatic caching of modules and build
+        # cache, speeding up builds. The cache key is dependent on the Go
+        # version and our go.sum contents.
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache: true
+
+      - name: Install dependencies
+        run: |
+          go install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0
+
+      - name: Build and test
+        run: |
+          go run build.go
+          go run build.go test
+
+      - name: Create packages
+        # Only run this for pushes to our branches, not on PRs. Uses secrets
+        # for code signing.
+        if: github.event_name == 'push'
+        run: |
```

Comment:
This feels a bit a bit weak.
I know that only we can push to "our" branches, but I'd felt more comfortable of this had some sort of branch name check etc, or something.

Reply:
Right - I'd even say we don't need any CI for random branches. So a `branches: ["main"]` condition at the top on the triggers could ensure that.

Reply:
But we still want to build on PRs to check that a given PR doesn't break the Windows build, don't we?

Reply:
There are two triggers: One for PR and one for push, which are independent. The branch restrictions is only needed on the push one.

Reply:
Yeah... I was thinking about that. We currently produce artifacts for any all builds on random branches, and that might be something we want to preserve? Otherwise we can of course easily restrict it to just builds on main. 

(I also have a much more elaborate extension of this PR, with matrix builds of go version x platform and package builds for all the things we usually build for...)

Reply:
Would it be possible to exfiltrate e.g signing certificate that way?

Reply:
I don't think so; PRs don't get access to secrets, so we'd have to approve and merge the exfiltrating code first. Probably you can do that with our current build setup, though...

Reply:
> Yeah... I was thinking about that. We currently produce artifacts for any all builds on random branches, and that might be something we want to preserve? Otherwise we can of course easily restrict it to just builds on main.
> 
> (I also have a much more elaborate extension of this PR, with matrix builds of go version x platform and package builds for all the things we usually build for...)

do random branch builds need to be signed?
don't think so.

Reply:
No, they really don't. GitHub actions has an "environments" concept where you can bind secrets to an environment and say that a certain branch should run in that environment, I think. I'll see if I can make the signing secrets specific to builds on main and release branches only.